### ISSUE TITLE
Mutation Testing

### ISSUE THREAD

**Issue by saralab** [stacks-network/stacks-core#4172]
Mutation Testing is a technique of evaluating the effectiveness of a series of tests by introducing small changes to the code (mutations) and checking if the tests can detect these small changes.

They are useful because they can show which functions are not tested properly and tell that their tests are missing, timing out or simply those functions are unviable to be tested.
```[tasklist]
### Independent Workflows
- [ ] https://github.com/stacks-network/stacks-core/issues/4176
- [ ] https://github.com/stacks-network/stacks-core/issues/4177
```

**Comment by ASuciuX** [comment]
Added a list with the workflows as I've split this into 2 PRs since the filter-pr could be merged and would be useful as soon as possible and waiting on the other one for the sync to be finished.
